// @surgeapi/node@0.33.0/resources/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhooks = exports.Verifications = exports.Users = exports.PhoneNumbers = exports.Messages = exports.Contacts = exports.Campaigns = exports.Blasts = exports.Accounts = void 0;
const tslib_1 = require("../internal/tslib.js");
tslib_1.__exportStar(require("./shared.js"), exports);
var accounts_1 = require("./accounts.js");
Object.defineProperty(exports, "Accounts", { enumerable: true, get: function () { return accounts_1.Accounts; } });
var blasts_1 = require("./blasts.js");
Object.defineProperty(exports, "Blasts", { enumerable: true, get: function () { return blasts_1.Blasts; } });
var campaigns_1 = require("./campaigns.js");
Object.defineProperty(exports, "Campaigns", { enumerable: true, get: function () { return campaigns_1.Campaigns; } });
var contacts_1 = require("./contacts.js");
Object.defineProperty(exports, "Contacts", { enumerable: true, get: function () { return contacts_1.Contacts; } });
var messages_1 = require("./messages.js");
Object.defineProperty(exports, "Messages", { enumerable: true, get: function () { return messages_1.Messages; } });
var phone_numbers_1 = require("./phone-numbers.js");
Object.defineProperty(exports, "PhoneNumbers", { enumerable: true, get: function () { return phone_numbers_1.PhoneNumbers; } });
var users_1 = require("./users.js");
Object.defineProperty(exports, "Users", { enumerable: true, get: function () { return users_1.Users; } });
var verifications_1 = require("./verifications.js");
Object.defineProperty(exports, "Verifications", { enumerable: true, get: function () { return verifications_1.Verifications; } });
var webhooks_1 = require("./webhooks.js");
Object.defineProperty(exports, "Webhooks", { enumerable: true, get: function () { return webhooks_1.Webhooks; } });
