// @surgeapi/node@0.33.0/resources/users.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { path } from "../internal/utils/path.mjs";
export class Users extends APIResource {
    /**
     * Creates a new User object.
     *
     * @example
     * ```ts
     * const user = await client.users.create(
     *   'acct_01j9a43avnfqzbjfch6pygv1td',
     *   { first_name: 'Brian' },
     * );
     * ```
     */
    create(accountID, body, options) {
        return this._client.post(path `/accounts/${accountID}/users`, { body, ...options });
    }
    /**
     * Retrieves a User object.
     *
     * @example
     * ```ts
     * const user = await client.users.retrieve(
     *   'usr_01j9dwavghe1ttppewekjjkfrx',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/users/${id}`, options);
    }
    /**
     * Updates an existing User object.
     *
     * @example
     * ```ts
     * const user = await client.users.update(
     *   'usr_01j9dwavghe1ttppewekjjkfrx',
     *   { first_name: 'Brian' },
     * );
     * ```
     */
    update(id, body, options) {
        return this._client.patch(path `/users/${id}`, { body, ...options });
    }
    /**
     * Deletes a user.
     *
     * Once a user has been deleted, they will no longer be permitted to access any of
     * the embedded components. Attempting to access a deleted user will return a 404
     * Not Found error.
     *
     * @example
     * ```ts
     * const user = await client.users.delete(
     *   'usr_01j9dwavghe1ttppewekjjkfrx',
     * );
     * ```
     */
    delete(id, options) {
        return this._client.delete(path `/users/${id}`, options);
    }
    /**
     * Provides a mechanism for having Surge create a signed token for embeds instead
     * of signing with your own signing key.
     *
     * @example
     * ```ts
     * const userTokenResponse = await client.users.createToken(
     *   'usr_01jymgdfrpec2asc5m0z3a6fr9',
     * );
     * ```
     */
    createToken(userID, body, options) {
        return this._client.post(path `/users/${userID}/tokens`, { body, ...options });
    }
}
